### PR TITLE
Merge tag Homebrew/1.1.11

### PR DESCRIPTION
**Pull request by bw-yuhogg**
Due to the magnitude of changes here, IMO we should not merge this for a while.

### CODE REVIEW

File: Library/Homebrew/cmd/cask.rb
Comment (bw-yuhogg) [inline Library/Homebrew/cmd/cask.rb]: @Linuxbrew/core we might want to put some sort of bail-out code here for Linux.

File: Library/Homebrew/extend/os/mac/diagnostic.rb
Comment (bw-yuhogg) [inline Library/Homebrew/extend/os/mac/diagnostic.rb:60]: FIXME: need to investigate this one and figure out why Circle is no longer mentioned.

File: Library/Homebrew/keg.rb
Comment (bw-yuhogg) [inline Library/Homebrew/keg.rb:211]: FIXME: not sure what's up with this one, have to look more closely.

File: Library/Homebrew/test/formula_spec.rb
Comment (maxim-belkin) [inline Library/Homebrew/test/formula_spec.rb]: should we move that `if OS.mac?` to `X11Requirement` which would require `XorgRequirement` on Linux platforms? This is a general question so it should not block this PR

File: Library/Homebrew/extend/os/mac/diagnostic.rb
Comment (maxim-belkin) [reply Library/Homebrew/extend/os/mac/diagnostic.rb:60]: Function name is `check_xcode_up_to_date`, so I would add ```return unless OS.mac?``` in the very beginning :)

Anyways, related commit is:

```patch
$ git show 6ac0de1b
commit 6ac0de1b40699042ea124813f2e953285efbd05f
Author: Mike McQuaid <mike@mikemcquaid.com>
Date:   Fri Feb 17 21:27:40 2017 +0000

    Revert "Skip Xcode outdated check on CircleCI"

diff --git a/Library/Homebrew/extend/os/mac/diagnostic.rb b/Library/Homebrew/extend/os/mac/diagnostic.rb
index 3e0c1d0..6d30cd2 100644
--- a/Library/Homebrew/extend/os/mac/diagnostic.rb
+++ b/Library/Homebrew/extend/os/mac/diagnostic.rb
@@ -53,8 +53,8 @@ module Homebrew
         return unless MacOS::Xcode.installed?
         return unless MacOS::Xcode.outdated?
 
-        # CI images are going to end up outdated so don't complain.
-        return if ENV["TRAVIS"] || ENV["CIRCLECI"]
+        # Travis CI images are going to end up outdated so don't complain.
+        return if ENV["TRAVIS"]
```

File: Library/Homebrew/test/formula_spec.rb
Comment (bw-yuhogg) [reply Library/Homebrew/test/formula_spec.rb]: Glad you asked. In the future, I want to rewrite the X* requirements to use the OS extension system Mike introduced upstream, in the same manner as my work on the Java and FUSE requirements.

Presumably, this would imply:

1. Moving the XQuartz logic to extend/os/mac
2. Moving the linuxbrew/xorg/xorg handling stuff to extend/os/linux
3.  (Only on Linuxbrew, not upstream) adding XorgRequirement to the compat file to point to X11Requirement

What do you think of that plan?

File: Library/Homebrew/extend/os/mac/diagnostic.rb
Comment (bw-yuhogg) [reply Library/Homebrew/extend/os/mac/diagnostic.rb:60]: Ah, I didn't even realize that this had nothing to do with Linuxbrew. I just saw "CIRCLECI" and assumed one of us put it there. But it's actually in extend/os/mac, so there's no action required on our end.

File: Library/Homebrew/test/formula_spec.rb
Comment (maxim-belkin) [reply Library/Homebrew/test/formula_spec.rb]: LGTM! As long as this work makes Linuxbrew and Homebrew codebase closer together, I'm 👍 on that

File: Library/Homebrew/extend/os/mac/diagnostic.rb
Comment (sjackman) [reply Library/Homebrew/extend/os/mac/diagnostic.rb:60]: Note that Linuxbrew currently does import `extend/os/mac` because we still use some functions in there. That definitely needs to be fixed. Linuxbrew should not import  `extend/os/mac`.

Comment (bw-yuhogg) [reply Library/Homebrew/extend/os/mac/diagnostic.rb:60]: 😱

File: Library/Homebrew/keg.rb
Comment (sjackman) [reply Library/Homebrew/keg.rb:211]: Looks to me that it no longer recurses into subdirectories (`*` instead of `**/*`) and it's considered non-empty if it contains a directory that its not itself empty (while ignoring Mac special files named `.DS_Store`).

File: Library/Homebrew/extend/os/mac/diagnostic.rb
Comment (maxim-belkin) [reply Library/Homebrew/extend/os/mac/diagnostic.rb:60]: 🍠

Comment (sjackman) [reply Library/Homebrew/extend/os/mac/diagnostic.rb:60]: The horror! The horror!

**Comment by bw-yuhogg** [comment]
@Linuxbrew/core all in favour of merging this to master (but not creating a new tag)?

It'd be easier for us to catch any remaining problems if we're all using it (hint hint...)

**Comment by sjackman** [comment]
:shipit: Go ahead and merge it to master if you think it's ready. Thanks for tackling this merge, Bob. Sounds like it was a tricky one. We can tag it as stable once we've had a couple of days to test it.

**Review by sjackman — APPROVED** [review]

**Comment by maxim-belkin** [comment]
I've played around with this one and further fixes require taking care of "... GCC requested but formula gcc not installed...", that is, it will require @sjackman :)

**Comment by sjackman** [comment]
> GCC requested but formula gcc not installed...

What commands are needed to reproduce that error?

**Comment by sjackman** [comment]
Merged, but not yet tagged.

**Comment by sjackman** [comment]
Thanks for all of this work, Bob!

**Comment by maxim-belkin** [comment]
> What commands are needed to reproduce that error?

```
brew tests
```

**Comment by sjackman** [comment]
I'm not seeing that failure with `brew tests`. In which environment, or was the test disabled?

**Pull request merged**
